File: src/main/java/xyz/phanta/cbtweaker/buffer/ingredient/IngredientProvider.java
package xyz.phanta.cbtweaker.buffer.ingredient;

import xyz.phanta.cbtweaker.integration.jei.ingredient.JeiIngredientProvider;

import java.util.function.Supplier;

public interface IngredientProvider<A, JA> extends JeiIngredientProvider<JA> {

    default void insertPeriodic(Supplier<A> acc) {
        // NO-OP
    }

    default boolean insertFinal(Supplier<A> acc) {
        return true;
    }

}

File: src/main/kotlin/st/evening/mc/cbtweaker/buffer/ingredient/IngredientProvider.kt
package st.evening.mc.cbtweaker.buffer.ingredient

import st.evening.mc.cbtweaker.compat.jei.ingredient.JeiIngredientProvider

interface IngredientProvider<A, JA> : JeiIngredientProvider<JA> {
    fun insertPeriodic(acc: Lazy<A>): Boolean = true

    fun insertFinal(acc: Lazy<A>): Boolean = true
}
